components/ConsolidatedFiles.tsx

```
@@ -0,0 +1,190 @@
+           hasMd: true,
+           hasJson: hasJson,
+         };
+      }).filter((file: ProcessedFile) => file.baseName !== 'invalid_name');
+
+      // Deduplicate based on baseName
+      const uniqueFilesMap = new Map<string, ProcessedFile>();
+      processedData.forEach(file => {
+          if (!uniqueFilesMap.has(file.baseName)) {
+              uniqueFilesMap.set(file.baseName, file);
+          } else {
+              const existing = uniqueFilesMap.get(file.baseName)!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)